packages/scheduler/src/forks/Scheduler.js

```diff
@@ -450,7 +450,7 @@ function shouldYieldToHost() {
       }
       // There's no pending input. Only yield if we've reached the max
       // yield interval.
-      return currentTime >= maxYieldInterval;
+      return currentTime >= Math.max(deadline, deadline - yieldInterval + maxYieldInterval);
```

Comment:
I think this is sufficient:

```suggestion
      return currentTime - deadline >= maxYieldInterval;
```